exercises/saddle-points/.meta/template.j2

```diff
@@ -0,0 +1,58 @@
+"""Tests for the saddle-points exercise
+
+Implementation note:
+The saddle_points function must validate the input matrix and raise a
+ValueError with a meaningful error message if the matrix turns out to be
+irregular.
+"""
+{%- import "generator_macros.j2" as macros with context -%}
+
+{% macro test_case(case) -%}
+    {%- set input = case["input"] -%}
+    {%- set expected = case["expected"] -%}
+    def test_{{ case["description"] | to_snake }}(self):
+        {% if input["matrix"][0] -%}
+        matrix = {{ input["matrix"] }}
+        {%- else -%} matrix = [] {% endif %}
+        {% if expected|length > 2 -%}
+        expected = {{ list_points(expected) }}
+        self.assertEqual({{ case["property"] | to_snake }}(matrix), expected)
+        {% else -%}
+        self.assertEqual(
+            {{- case["property"] | to_snake }}(matrix), {{ list_points(expected) }})
+        {%- endif %}
```

Comment:
To make maintaining the template easier, all non-error tests may directly embed `list_points(expected)` in the `assertEqual` call (currently inside the `{% else %}` here). No need to check length of `expected`.

Reply:
Cleaned up.